dataservice/api/biospecimen/models.py

```diff
@@ -98,11 +100,14 @@ class Biospecimen(db.Model, Base):
     dbgap_consent_code = db.Column(db.Text(),
                                    doc='Consent classification code from dbgap'
                                    )
-    genomic_files = db.relationship(GenomicFile,
-                                    cascade="all, delete-orphan",
-                                    backref=db.backref(
-                                        'biospecimen', lazy=True),
-                                    doc='genomic files this biospecimen')
     diagnoses = db.relationship(Diagnosis,
                                 backref=db.backref('biospecimen',
                                                    lazy=True))
+    genomic_files = association_proxy(
+        'biospecimen_genomic_files', 'genomic_file',
+        creator=lambda genomic_file:
+        BiospecimenGenomicFile(genomic_file=genomic_file))
+
+    biospecimen_genomic_files = db.relationship(BiospecimenGenomicFile,
```

Comment:
does this really need to be defined? I don't think we'll need to access it directly.

Reply:
I think we do need this defined bc we need the cascade all delete orphan part. We want it so that when you delete a Biospecimen, it will delete the rows in the association table that reference that biospecimen. Usually you don't have to specify the cascade's manually for many-to-many, but since we're using sqlalchemy association object rather than association table to define the association table we have to specify the cascade manually on both the biospecimen and genomic file tables. Alternatively, we might be able to define both the cascades in the association obj class BiospecimenGenomicFile .. but its prob fine the way it is